src/ipc-handlers.ts

```diff
@@ -1081,6 +1070,95 @@ END` );
 	}
 }
 
+export async function openAppAtPath(
+	_event: IpcMainInvokeEvent,
+	editorKey: SupportedEditor,
+	filePath: string
+): Promise< void > {
+	const platform = process.platform;
+	const editor = supportedEditorConfig[ editorKey ];
+
+	if ( ! editor ) {
+		throw new Error( `Editor ${ editorKey } not found or not supported` );
+	}
+
+	if ( platform === 'darwin' ) {
+		return promiseExec( `open -b ${ editor.bundleId } "${ filePath }"` );
+	}
+
+	if ( platform === 'win32' ) {
+		const editorPath = await winFindEditorPath( editorKey );
+		if ( ! editorPath ) {
+			// Fall back to using openURL if no editor path is found
+			return openURL( _event, editor.url( filePath ) );
+		}
+
+		return promiseExec( `"${ editorPath }" "${ filePath }"` );
+	}
+
+	throw new Error( `Platform ${ platform } is not supported` );
+}
+
+/**
+ * Expands Windows environment variables in a path
+ */
+function winExpandPath( path: string ): string {
+	return path.replace( /%([^%]+)%/g, ( _, n ) => process.env[ n ] || '' );
+}
+
+/**
+ * Finds the executable path for a given editor on Windows
+ */
+async function winFindEditorPath( editorKey: SupportedEditor ): Promise< string | null > {
+	const editor = supportedEditorConfig[ editorKey ];
+	if ( ! editor || ! editor.winPaths ) {
+		return null;
+	}
+
+	for ( const possiblePath of editor.winPaths ) {
+		const expandedPath = winExpandPath( possiblePath );
+
+		// Handle wildcards in paths (for JetBrains Toolbox installations)
+		if ( expandedPath.includes( '*' ) ) {
+			const basePath = nodePath.dirname( expandedPath );
+			const pattern = nodePath.basename( expandedPath );
+
+			try {
+				const files = await fs.promises.readdir( basePath );
+				const matchingFiles = files.filter( ( f ) =>
+					f.match( new RegExp( pattern.replace( '*', '.*' ) ) )
+				);
+
+				for ( const file of matchingFiles ) {
+					const fullPath = nodePath.join( basePath, file );
+					if (
+						await fs.promises
+							.access( fullPath )
+							.then( () => true )
+							.catch( () => false )
+					) {
+						return fullPath;
+					}
+				}
+			} catch ( error ) {
+				// Skip if directory doesn't exist
+				continue;
+			}
+		} else {
```

Comment:
Is this needed? None of the paths in `src/modules/user-settings/lib/editor.ts` contain a wildcard character.

Reply:
I was experimenting with recognizing Jetbrains IDEs in their versioned folders, and for these, I included a `*` in the path to match them. Later I removed the wildcards from the paths because I thought Jetbrains doesn't use the versioned folders anymore, but that seem to only be the case when installed for the current user only.

As @gavande1 [pointed out below](https://github.com/Automattic/studio/pull/1346#discussion_r2079276669), this might be needed for system-wise installations of the Jetbrains IDEs after all. I'll revert that path to include a wildcard.

Reply:
I updated the matching logic in 7ca0c5a4e4a62b531c599c87f278aaad8681c357